DeepCat.js

```diff
@@ -363,26 +366,49 @@
 	}
 
 	function addSearchFormHint() {
-		var hintBox = '<div style="margin-top: 1em; margin-bottom: 1em;">'
-						+ '<div id="deepcat-hintbox" style="background: #8af; padding: .75em; width: 75%; display: none">'
+		var hintBox = '<div id="deepcat-hintbox">'
+						+ '<img id="deepcat-info-img" src="//upload.wikimedia.org/wikipedia/commons/thumb/1/1d/Information_icon4.svg/40px-Information_icon4.svg.png" style="display: none;"/>  '
+						+ '<div>'
 							+ mw.msg( 'deepcat-hintbox-text' )
-							+ '<div style="text-align: right;">'
-								+ '<button id="deepcat-hint-hide">' + mw.msg( 'deepcat-hintbox-close' ) + '</button>'
-						+ '</div></div></div>';
+							+ '&nbsp;<a id="deepcat-hint-hide">' + mw.msg( 'deepcat-hintbox-close' ) + '</a>'
+						+ '</div></div>';
 		$( '#search' ).after( hintBox );
 		$( '#deepcat-hint-hide' ).on( 'click', hideHints );
 	}
 
+	function addSmallFormHint() {
+		var smallHintBox = '<div id="deepcat-smallhint">'
+						+ '<img id="deepcat-smallhint-hide" title="' + mw.msg( 'deepcat-hintbox-close' ) + '" src="https://upload.wikimedia.org/wikipedia/commons/4/44/Curation_bar_icon_close.png">'
+						+ mw.msg( 'deepcat-hintbox-small' )
+						+ '</div>';
+		$( '#searchform' ).after( smallHintBox );
+		$( '#deepcat-smallhint-hide' ).on( 'click', hideHints );
+	}
+
 	function hasHintCookie() {
 		return mw.cookie.get( "-deepcat-hintboxshown" ) == makeHintboxCookieToken( mw.msg( 'deepcat-hintbox-text' ) );
 	}
 
 	function hideHints() {
 		shouldHideHints = true;
+
 		$( '#deepcat-hintbox' ).hide();
+		$( '#deepcat-smallhint' ).hide();
+		enableImeAndSuggestions();
+
 		mw.cookie.set( "-deepcat-hintboxshown", makeHintboxCookieToken( mw.msg( 'deepcat-hintbox-text' ) ), { 'expires': 60 * 60 * 24 * 7 * 4 /*4 weeks*/ } );
 	}
 
+	function disableImeAndSuggestions() {
+		$( '.suggestions' ).css( 'z-index', -1 );
+		$( '.imeselector' ).css( 'z-index', -1 );
```

Comment:
IME can be disabled by adding the class name `noime` to the input element. I do not know what "suggestions" refers to. The browsers auto-completions can be disabled with `autocomplete="off"` on the element or the form.